File: actions/GetMatrix.py
def GetMatrix(r, colCount):
    """
    функция,которая делит вектор-сигнала
    на колл-во-colCount длинну фильтра(3),
    из вектора получаем матрицу
    """
    array = r
    m = len(r)
    matrix = [[]]
    rowIndex = 0
    for ii in range(m):
        matrix[rowIndex].append(array[ii])
        if (ii + 1) % colCount == 0 and (ii + 1) != m:
            rowIndex += 1
            matrix.append([])
    return matrix

File: matlab/GetMatrix.m
function [ matrix ] = GetMatrix(r,colCount )
%функция,которая делит вектор-сигнала на колл-во-colCount длинну
%фильтра(3), из вектора получаем матрицу
array=r;
m = length(r);

  rowIndex = 1;
  colIndex = 1;
  for ii = 1:m 

    matrix(rowIndex,colIndex) = array(ii);
    colIndex=colIndex+1;
    if mod(ii,colCount)==0  
       rowIndex=  rowIndex+1;
        colIndex = 1;
    end;
  end;  

end

